File: simjaesung/week9/1071.java
class Solution {
    public String gcdOfStrings(String str1, String str2) {
        int len1 = str1.length();
        int len2 = str2.length();
        int gcd = gcd(len1,len2);
        String repeatString = str1.substring(0,gcd);
        if(checkRepeat(gcd,str1,repeatString) && checkRepeat(gcd,str2,repeatString)){
            return repeatString;
        }else{
            return "";
        }
    }

    public int gcd(int a, int b){
        while(b != 0){
            int r = a % b;
            a = b;
            b = r;
        }
        return a;
    }

    public boolean checkRepeat(int gcd, String s, String repeatString){
        for(int i = 0; i< s.length(); i += gcd){
            if(!s.substring(i,i + gcd).equals(repeatString)){
                return false;
            }
        }
        return true;
    }
}

File: yiseungyun/week9/1071.js
/**
 * @param {string} str1
 * @param {string} str2
 * @return {string}
 */

var gcdOfStrings = function(str1, str2) {
  const length1 = str1.length;
  const length2 = str2.length;
  let gcdNum;

  if (str1+str2 !== str2+str1) return "";

  if (length1 > length2) {
      gcdNum = gcd(length1, length2);
  } else {
      gcdNum = gcd(length2, length1);
  }

  const subStr = str1.slice(0, gcdNum);

  return subStr;
};

const gcd = (a, b) => {
  if (b === 0) return a;
  else return gcd(b, a%b);
}